src/client/Client.cc

```diff
@@ -1973,6 +1974,11 @@ int Client::make_request(MetaRequest *request,
       session = mds_sessions.at(mds);
     }
 
+    if (feature_needed != ULONG_MAX && !session->mds_features.test(feature_needed)) {
+      request->abort(-CEPHFS_ENODATA);
```

Comment:
oops: this should be `EOPNOTSUPP` but `_getvxattr` below should convert to `ENODATA`.

Reply:
Makes sense. Done!